Remind about AutoNonVariableTypeMode in error message.

ailzhang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51655 Remind about AutoNonVariableTypeMode in error message.**

Differential Revision: [D26228508](https://our.internmc.facebook.com/intern/diff/D26228508)

Commit:
Remind about AutoNonVariableTypeMode in error message.

[ghstack-poisoned]

ailzhang commented:
Improve error message to give more hints about skipping autograd (which is a common reason users hit this error).

xta0 reviewed (approved):
LGTM. Thanks for adding the message!

bhosmer commented on aten/src/ATen/core/boxing/KernelFunction.cpp:
```diff
@@ -23,6 +23,8 @@ void ambiguous_autogradother_kernel(OperatorKernel*, const OperatorHandle& op, S
     op.operator_name(), " has kernels registered to both Math and a backend mapped to AutogradOther. "
     "This makes the backend kernel unreachable (see Note [Ambiguity in AutogradOther kernel]). "
+    "If you only want to run inference instead of training, add `at::AutoNonVariableTypeMode guard(true);` "
+    "before model.forward().\n"
```
I'd actually move this new info to the end of the text, so the "If it's intended..." part relates more clearly to the AutogradOther kernel mentioned above.

bhosmer replied:
Also, is it a problem that the advice is C++ specific?

bhosmer reviewed (approved):
Definitely nice to have the added info - couple things inline, gimme a shout if you wanna talk C++ vs Python messaging 😁

Commit:
Update on "Remind about AutoNonVariableTypeMode in error message."


Differential Revision: [D26228508](https://our.internmc.facebook.com/intern/diff/D26228508)

[ghstack-poisoned]
